runtime/integration-tests/src/evm.rs

```diff
@@ -706,3 +731,316 @@ fn test_evm_accounts_module() {
+			assert_eq!(
+				Currencies::total_issuance(CurrencyId::DexShare(
+					DexShare::Erc20(erc20_address_0()),
+					DexShare::Token(NATIVE_TOKEN_SYMBOL),
+				)),
+				0
+			);
+			assert_eq!(
+				Currencies::free_balance(
+					CurrencyId::DexShare(DexShare::Erc20(erc20_address_0()), DexShare::Token(NATIVE_TOKEN_SYMBOL)),
+					&AccountId::from(ALICE)
+				),
+				0
+			);
+			assert_eq!(
+				Currencies::free_balance(
+					CurrencyId::DexShare(DexShare::Erc20(erc20_address_0()), DexShare::Token(NATIVE_TOKEN_SYMBOL)),
+					&MockAddressMapping::get_account_id(&alice_evm_addr())
+				),
+				0
+			);
+
+			// CurrencyId::DexShare(Erc20, ACA)
+			<EVM as EVMTrait<AccountId>>::set_origin(EvmAddressMapping::<Runtime>::get_account_id(&alice_evm_addr()));
+			assert_ok!(Dex::add_provision(
+				Origin::signed(EvmAddressMapping::<Runtime>::get_account_id(&alice_evm_addr())),
+				CurrencyId::Erc20(erc20_address_0()),
+				NATIVE_CURRENCY,
+				100 * dollar(NATIVE_CURRENCY),
+				1000 * dollar(NATIVE_CURRENCY),
+			));
+			assert_ok!(Dex::end_provisioning(
+				Origin::signed(AccountId::from(BOB)),
+				CurrencyId::Erc20(erc20_address_0()),
+				NATIVE_CURRENCY,
+			));
+			assert_eq!(
+				Dex::get_liquidity_pool(CurrencyId::Erc20(erc20_address_0()), NATIVE_CURRENCY,),
+				(110 * dollar(NATIVE_CURRENCY), 1100 * dollar(NATIVE_CURRENCY))
+			);
+
+			// The order of dexshare is related
+			assert_eq!(
+				Currencies::total_issuance(CurrencyId::DexShare(
+					DexShare::Erc20(erc20_address_0()),
+					DexShare::Token(NATIVE_TOKEN_SYMBOL),
+				)),
+				0
+			);
+			assert_eq!(
+				Currencies::total_issuance(CurrencyId::DexShare(
+					DexShare::Token(NATIVE_TOKEN_SYMBOL),
+					DexShare::Erc20(erc20_address_0()),
+				)),
+				2200 * dollar(NATIVE_CURRENCY)
+			);
+			assert_eq!(
+				Currencies::total_issuance(lp_erc20_aca()),
+				2200 * dollar(NATIVE_CURRENCY)
+			);
+
+			// set_global_fee_swap_path
+			assert_ok!(TransactionPayment::set_global_fee_swap_path(
+				Origin::root(),
+				vec![CurrencyId::Erc20(erc20_address_0()), NATIVE_CURRENCY]
+			));
+			assert_ok!(Currencies::update_balance(
+				Origin::root(),
+				MultiAddress::Id(TreasuryAccount::get()),
+				NATIVE_CURRENCY,
+				(100 * dollar(NATIVE_CURRENCY)).try_into().unwrap()
+			));
+
+			assert_eq!(Currencies::free_balance(NATIVE_CURRENCY, &sub_account), 0);
+			assert_eq!(
+				Currencies::free_balance(CurrencyId::Erc20(erc20_address_0()), &sub_account),
+				0
+			);
+
+			assert_ok!(TransactionPayment::enable_charge_fee_pool(
+				Origin::root(),
+				CurrencyId::Erc20(erc20_address_0()),
+				5 * dollar(NATIVE_CURRENCY),
+				Ratio::saturating_from_rational(35, 100).saturating_mul_int(dollar(NATIVE_CURRENCY)),
+			));
+
+			assert_eq!(
+				Currencies::free_balance(NATIVE_CURRENCY, &sub_account),
+				5 * dollar(NATIVE_CURRENCY)
+			);
+			// erc20 minimum_balance is 0
+			assert_eq!(
+				Currencies::free_balance(CurrencyId::Erc20(erc20_address_0()), &sub_account),
+				0
```

Comment:
is it dex pair not register to asset-registry cause ED of dex = 0?

https://github.com/AcalaNetwork/Acala/blob/8560555fb9572517dafcc6557f4ae1879fc99e1d/runtime/karura/src/lib.rs?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L788-L791